src/Entity/OgMembership.php

```diff
@@ -319,6 +319,12 @@ public function preSave(EntityStorageInterface $storage) {
       throw new \LogicException(sprintf('Entity type %s with ID %s is not an OG group.', $entity_type_id, $group->id()));
     }
 
+    foreach ($this->getRoles() as $role) {
+      if ($role->getName() == OgRoleInterface::ANONYMOUS) {
+        throw new \LogicException('A group member role is required for creating a membership.');
```

Comment:
Maybe change the name to something more explicit like `Cannot save an OgMembership with reference to a non-member role.`